Assets/Scripts/Augment/BulletModifiers/KnockbackOnShotModifier.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class KnockbackOnShotModifier: MonoBehaviour, ProjectileModifier
+{
+    [SerializeField]
+    private float pushPower;
+
+    [SerializeField]
+    private float radius;
+
+    [SerializeField]
+    private KnockbackEffect knockBackScript;
+
+    private float bulletAmount = 1f;
+
+    [SerializeField]
+    private Transform[] knockbackNormal;
+
+    private PlayerManager source;
+
+
+    public void Attach(ProjectileController projectile)
+    {
+        projectile.OnProjectileInit += KnockAwayOnShot;
+
+        var bulletController = projectile.gameObject.GetComponent<BulletController>();
+        bulletAmount = bulletController == null ? 1f : bulletController.bulletsPerShot;
+        
+        if (bulletAmount >= 5)
+        {
+            bulletAmount -= 4;
```

Comment:
You might want to adjust this to adjust more dynamically. We could get weapons with lots more bullets per shot eventually, and have you tested with 2-4 bullets?

Reply:
Replaced with `(pushPower / bulletAmount) * (1f + (float) Math.Log10(bulletAmount))`. This seems to solve the issue at least for shotgun / bulletamounts, although I'm unsure how we should resolve the "propeller" effect when combining solar body and normal gun barrel.

Reply:
We'll have to adjust the solar body anyways, but perhaps you could assist knockback effect based on rate of fire?